File: c4/c4.ts
function createXmasTree(height: number, ornament: string): string {
  let tree: string[] = [];
  for (let i = 1; i <= height; i++) {
    const ornaments = ornament.repeat(i * 2 - 1);
    const spaces = '_'.repeat(height - i);
    tree.push(spaces + ornaments + spaces);
  }
  const trunk = '_'.repeat(height - 1) + '#' + '_'.repeat(height - 1);
  tree.push(trunk);
  tree.push(trunk);
  return tree.join('\n');
}

File: c4/c4.py
def create_xmas_tree(height, ornament):
    tree = []
    for i in range(1, height + 1):
        ornaments = ornament * (i * 2 - 1)
        spaces = "_" * (height - i)
        tree.append(spaces + ornaments + spaces)

    trunk = "_" * (height - 1) + "#" + "_" * (height - 1)
    tree.append(trunk)
    tree.append(trunk)
    return "\n".join(tree)
